app/uk/gov/hmrc/transitmovements/services/ObjectStoreService.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovements.services
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import cats.data.EitherT
+import com.google.inject.ImplementedBy
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.objectstore.client.play.PlayObjectStoreClient
+import uk.gov.hmrc.objectstore.client.play.Implicits._
+import uk.gov.hmrc.transitmovements.services.errors.ObjectStoreError
+import uk.gov.hmrc.objectstore.client.Path
+import uk.gov.hmrc.transitmovements.models.ObjectStoreURI
+
+import javax.inject._
+import scala.concurrent._
+import scala.util.control.NonFatal
+
+@ImplementedBy(classOf[ObjectStoreServiceImpl])
+trait ObjectStoreService {
+
+  def getObjectStoreFile(
+    objectStoreURI: ObjectStoreURI
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): EitherT[Future, ObjectStoreError, Source[ByteString, _]]
+
+}
+
+@Singleton
+class ObjectStoreServiceImpl @Inject() (client: PlayObjectStoreClient) extends ObjectStoreService {
+
+  override def getObjectStoreFile(
+    objectStoreURI: ObjectStoreURI
+  )(implicit ec: ExecutionContext, hc: HeaderCarrier): EitherT[Future, ObjectStoreError, Source[ByteString, _]] = {
+    val objectStoreFilePath = if (objectStoreURI.value.contains("/")) objectStoreURI.value.split("/", 2).apply(1) else objectStoreURI.value
```

Comment:
This line feels incredibly opaque to me. What is this line doing?

Reply:
We will getting objectStoreURI like common-transit-convention-traders/movements/:movementID ..... So with that we need to remove common-transit-convention-traders service part in order to get the file. I have tested this with Object store stub and looks good. When I am including common-transit-convention-traders with get request I am getting 404 File not Found. For reference please check: https://github.com/hmrc/object-store-client#get-object

Reply:
Right -- but that's not obvious to someone reading the code itself. This line is actually making an assumption about the URI provided -- and a valid one at that, but by just reading the code, without prior knowledge I don't know what it is. If/when you look at it in six months, or if Live Services touch it, that knowledge may not be there.

I'm still nervous about this because of what you're doing if the URI doesn't contain a `/`. If that happens, we know things have just gone plain wrong anyway. Additionally, we know that the URI must **always** be for `common-transit-convention-traders`, right?

So, I propose something a little tighter, something like this on `ObjectStoreURI`:

```
object  ObjectStoreURI {

    // The URI consists of the service name in the first part of the path, followed
    // by the location of the object in the context of that service. As this service
    // targets common-transit-convention-traders' objects exclusively, we ensure
    // the URI is targeting that context. This regex ensures that this is the case.
    val expectedUriPattern = "^common-transit-convention-traders/(.+)$".r

}

case class ObjectStoreURI(value: String) extends AnyVal {

    def path: Option[String] = value match {
        case ObjectStoreURI.expectedUriPattern(path) => Some(path)
        case _ => None
    }
}
```

Of course, this logic can be in the service too. Not bothered where it ends up. Then, you can handle the path you've just extracted easily.

Reply:
Done the changes as suggested above